frontend/src/metabase/visualizations/visualizations/SmartScalar/SmartScalar.jsx

```diff
@@ -111,14 +114,15 @@ export function SmartScalar({
   const dateColumn = cols[dimensionIndex];
   const dateColumnSettings = settings.column(dateColumn) ?? {};
 
-  const lastPeriod = unit
-    ? formatDateTimeRangeWithUnit([lastDate], unit, {
-        compact: true,
-      })
-    : formatValue(lastDate, {
-        ...dateColumnSettings,
-        column: dateColumn,
-      });
+  const lastPeriod =
+    !unit || queryType === "native"
+      ? formatValue(lastDate, {
+          ...dateColumnSettings,
+          column: dateColumn,
+        })
+      : formatDateTimeRangeWithUnit([lastDate], unit, {
+          compact: true,
+        });
```

Comment:
Matches this from the original PR - https://github.com/metabase/metabase/pull/38344/files#diff-195cb3e525a65af942309b028c271969b37c05448843666c178cb3708c266004R520-R535